components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -84,7 +94,14 @@ public void completedServerStartup() {
                 new APIMgtGatewayCacheMessageListener());
         jmsTransportHandlerForEventHub
                 .subscribeForJmsEvents(APIConstants.TopicNames.TOPIC_NOTIFICATION, new GatewayJMSMessageListener());
+    }
 
+    private void deployAPIsInSyncMode() {
+        isAPIsDeployedInSyncMode = deployArtifactsAtStartup();
+        if (!isAPIsDeployedInSyncMode) {
```

Comment:
no need to have if condition since it decided to go in sync.

Reply:
This to check Every API is deployed or not . https://github.com/wso2/carbon-apimgt/pull/8945#discussion_r451267177